src/journal/journald-server.c

```diff
@@ -573,7 +571,8 @@ static int server_do_rotate(
                                                          "Failed to create new %s journal: %m", name);
         }
 
-        server_add_acls(*f, uid);
+        if (uid > 0)
+                server_add_acls(*f, uid);
```

Comment:
i am not convinced about moving checks like this from the callee to the caller.

We typically do it the exact opposite way: make the callsites dumb, make the call itself smart with as many early exits as necessary. It's more robust that way, and it is easier to read, since the entry conditions are very clearly documented that way.

Reply:
OTOH calling server_add_acls() for the system journal can be seen as a mistake since perms of the system journal is not supposed to be modified to allow  users to access it. Hence making the function abort for the system journal seems to make sense in this case.

Reply:
sorry, but this is not our usual preferred coding style...

I would follow your logic if entirely distinct object types are passed here, that we'd assert() on the wrong one. but that's not the case here: it's a minor distinction if this is per-user journal file or a per-system one. hence please encapsulate such tests inside the functions.